core/drivers/imx_wdog.c

```diff
@@ -124,29 +112,34 @@ static TEE_Result imx_wdog_base(vaddr_t *wdog_vbase)
 	}
 
 	/* search the first usable wdog */
-	for (i = 0; i < ARRAY_SIZE(wdog_path); i++) {
-		off = fdt_path_offset(fdt, wdog_path[i]);
-		if (off < 0)
-			continue;
-
-		st = _fdt_get_status(fdt, off);
-		if (st & DT_STATUS_OK_SEC)
+	for (i = 0; i < ARRAY_SIZE(dt_wdog_match_table); i++) {
+		match = dt_wdog_match_table[i];
+		off = 0;
+		while (off >= 0) {
+			off = fdt_node_offset_by_compatible(fdt, off, match);
+			if (off > 0) {
+				st = _fdt_get_status(fdt, off);
+				if (st & DT_STATUS_OK_SEC) {
+					DMSG("Wdog found at %u", off);
+					found_off = off;
+					break;
+				}
+			}
+		}
+		if (found_off)
 			break;
+		else
+			DMSG("%s not found in DTB", dt_wdog_match_table[i]);
 	}
 
-	if (i == ARRAY_SIZE(wdog_path))
+	if (!found_off) {
+		EMSG("No Watchdog found in DTB\n");
 		return TEE_ERROR_ITEM_NOT_FOUND;
+	}
 
-	DMSG("path: %s\n", wdog_path[i]);
-
-	ext_reset = dt_have_prop(fdt, off, "fsl,ext-reset-output");
-
-	pbase = _fdt_reg_base_address(fdt, off);
-	if (pbase == (paddr_t)-1)
-		return TEE_ERROR_ITEM_NOT_FOUND;
+	ext_reset = dt_have_prop(fdt, found_off, "fsl,ext-reset-output");
 
-	sz = _fdt_reg_size(fdt, off);
-	if (sz < 0)
+	if(dt_map_dev(fdt, found_off, &pbase, &sz) < 0)
```

Comment:
```if (dt_map_dev(fdt, found_off, &pbase, &sz) < 0)```

Reply:
Shippable also reports a `ssize_t`/`size_t` issue here. I'll add the space than.